File: doc-examples/example-java/src/test/java/example/ProductTest.java
package example;

import java.io.IOException;

import io.micronaut.serde.ObjectMapper;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
public class ProductTest {

    @Test
    void testSerDeser(ObjectMapper objectMapper) throws IOException {
        final String result = objectMapper.writeValueAsString(new Product("Apple", 10));
        assertEquals(
                "{\"p_name\":\"Apple\",\"p_quantity\":10}",
                result
        );
        final Product product = objectMapper.readValue(result, Product.class);
        assertEquals(
                "Apple",
                product.getName()
        );
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/ProductTest.kt
package example

import io.micronaut.serde.ObjectMapper
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautTest
class ProductTest {

    @Test
    fun testSerDeser(objectMapper: ObjectMapper) {
        val result = objectMapper.writeValueAsString(Product("Apple", 10))
        assertEquals(
            "{\"p_name\":\"Apple\",\"p_quantity\":10}",
            result
        )
        val product = objectMapper.readValue(result, Product::class.java)!!
        assertEquals(
            "Apple",
            product.name
        )
    }
}
